src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+This is far better from what we had before. There is still room for improvement though. Right now we don't handle any errors. In a real application errors are unavoidable. Let's say that the service that we call to retrieve offers has a long response time. We can simulate it by using a mock `/timeout`. It responds with the same data as `/offers` but it takes three seconds to do so. Swap the endpoint in the `fetch-offers` action:
+
+```hocon
+fetch-offers {
+  factory = http
+  config.endpointOptions {
+    path = /timeout
+    domain = webapi
+    port = 8080
+  }
+}
+```
+
+When we load the page now it should take about three seconds. If we open the Knot.x extension we can immediatly spot on the fragment list that the second fragment executed for a long time. When we open the fragments performance view we can clearly see that the problem lays in the composite and specificaly in the `fetch-offers` node.
+
+What if such a long time is unacceptable? One approach would be to define timeout in the `fetch-offers` action. However, we can do better. Knot.x provides us with a built-in implementation of a [Circuit Breaker Pattern](https://en.wikipedia.org/wiki/Circuit_breaker_design_pattern). It does so by a [Circuit Breaker Factory](TODO-find-a-link-to-documentation) that acts as a wrapper. Let's see how it can be configured:
+
+```hocon
+fetch-offers-cb {
+  factory = cb
+  doAction = fetch-offers
+
+  config.circuitBreakerOptions {
+    maxFailures = 3
+    timeout = 500
+    resetTimeout = 20000
+  }
+}
+```
+
+The circuit breaker calls the `fetch-offers` action inside of itself (`doAction` property). If the action failed then the circuit breaker assumes something is wrong and won't actually call the action next time it's invoked. It'll just assume failure instead.
+
+We defined the following properties:
+- `maxFailures` - number of failures it takes for circuit breaker to assume the action is not working
+- `timeout` - the circuit breaker will assume failure after this time
+- `resetTimeout` - how long the circuit breaker should wait until it'll allow the action to execute again
+
+Now we need to adjust the task. This is how it should look like now:
+
+```
+payments-task {
+  action = fetch-user-info
+
+  on._success {
+    actions = [
+      {
+        action = fetch-payment-providers
+      },
+      {
+        action = fetch-offers-cb
+      }
+    ]
+
+    on {
+      _success {
+        action = te-pebble
+      }
+      _error {
+        action = te-pebble
+      }
+    }
+  }
+}
+```
+
+Firstly, we call `fetch-offers-cb` instead of `fetch-offers` in the composite. That's because the `fetch-offers` action will be called inside of `fetch-offers-cb`. Also, since we expect the task to fail, the composite should handle the `_error` transition. It will be called if any subtask ends with anythink different then success. In this case we still want to render the page so we call the `te-pebble` action. 

```

Comment:
```suggestion
First, we call `fetch-offers-cb` instead of `fetch-offers` in the composite. That's because the `fetch-offers` action will be called inside of `fetch-offers-cb`. Also, since we expect the task to fail, the composite should handle the `_error` transition. It will be called if any subtask ends with anything different than success. In this case, we still want to render the page so we call the `te-pebble` action. 
```